site/content/docs/02-template-syntax.md

```diff
@@ -0,0 +1,592 @@
+	<option value="Cheese">Cheese</option>
+	<option value="Guac (extra)">Guac (extra)</option>
+</select>
+```
+
+---
+
+When the value of an `<option>` matches its text content, the attribute can be omitted.
+
+```html
+<select multiple bind:value={fillings}>
+	<option>Rice</option>
+	<option>Beans</option>
+	<option>Cheese</option>
+	<option>Guac (extra)</option>
+</select>
+```
+
+---
+
+Media elements (`<audio>` and `<video>`) have their own set of bindings — four *readonly* ones...
+
+* `duration` (readonly) — the total duration of the video, in seconds
+* `buffered` (readonly) — an array of `{start, end}` objects
+* `seekable` (readonly) — ditto
+* `played` (readonly) — ditto
+
+...and three *two-way* bindings:
+
+* `currentTime` — the current point in the video, in seconds
+* `paused` — this one should be self-explanatory
+* `volume` — a value between 0 and 1
+
+```html
+<video
+	src={clip}
+	bind:duration
+	bind:buffered
+	bind:seekable
+	bind:played
+	bind:currentTime
+	bind:paused
+	bind:volume
+></video>
+```
+
+---
+
+Block-level elements have readonly `clientWidth`, `clientHeight`, `offsetWidth` and `offsetHeight` bindings, measured using a technique similar to [this one](http://www.backalleycoder.com/2013/03/18/cross-browser-event-based-element-resize-detection/).
+
+```html
+<div bind:offsetWidth={width} bind:offsetHeight={height}>
+	<Chart {width} {height}/>
+</div>
+```
+
+---
+
+To get a reference to a DOM node, use `bind:this`.
+
+```html
+<script>
+	import { onMount } from 'svelte';
+
+	let canvasElement;
+
+	onMount(() => {
+		const ctx = canvasElement.getContext('2d');
+		drawStuff(ctx);
+	});
+</script>
+
+<canvas bind:this={canvasElement}></canvas>
+```
+
+
+### Component bindings
+
+* `bind:property={value}`
+
+You can bind to component props using the same mechanism.
+
+```html
+<Keypad bind:value={pin}/>
+```
+
+---
+
+Components also support `bind:this`, allowing you to interact with component instances programmatically.
+
+(Note that we can do `{cart.empty}` rather than `{() => cart.empty()}`, since there is no `this` inside a component's `<script>` block.)
+
+```html
+<ShoppingCart bind:this={cart}/>
+
+<button on:click={cart.empty}>
+	Empty shopping cart
+</button>
+```
+
+
+### Classes
+
+* `class:name={value}`
+* `class:name`
+
+---
+
+A `class:` directive provides a shorter way of toggling a class on an element.
+
+```html
+<!-- These are equivalent -->
+<div class="{active ? 'active' : ''}">...</div>
+<div class:active={active}>...</div>
+
+<!-- Shorthand, for when name and value match -->
+<div class:active>...</div>
+```
+
+
+### Actions
+
+* `use:action`
+* `use:action={parameters}`
+
+```js
+action = (node: HTMLElement, parameters: any) => {
+	update?: (parameters: any) => void,
+	destroy: () => void
+}
+```
+
+---
+
+Actions are functions that are called when an element is created. They must return an object with a `destroy` method that is called after the element is unmounted:
```

Comment:
Do they actually _have_ to return a `destroy`? There's not an implicit no-op when an element is unmounted?

Reply:
I thought they did but they don't! Nice